zproject/test_settings.py

```diff
@@ -126,7 +126,9 @@ def set_loglevel(logger_name, level) -> None:
 
 # These settings are set dynamically in `zerver/lib/test_runner.py`:
 TEST_WORKER_DIR = ''
-LOCAL_UPLOADS_DIR = ''
+# Don't unintentionally override a set value.
+if "LOCAL_UPLOADS_DIR" not in os.environ:
+    LOCAL_UPLOADS_DIR = ''
```

Comment:
This architecture is weird and convoluted.  You should just set `LOCAL_UPLOADS_DIR` based on the environment here if present, and set the value in an environment variable in the calling code above.

Reply:
Good to know. That's what I had originally had done, but I wasn't sure if we wanted to keep `LOCAL_UPLOADS_DIR` as being set to `''`, thus the weirdness.